matrix_client/crypto/olm_device.py

```diff
@@ -54,3 +54,41 @@ def sign_json(self, json):
             json['unsigned'] = unsigned
 
         return json
+
+    def verify_json(self, json, user_key, user_id, device_id):
+        """Verifies a signed key object's signature.
+
+        The object must have a 'signatures' key associated with an object of the form
+        `user_id: {key_id: signature}`.
+
+        Args:
+            json (dict): The JSON object to verify.
+            user_key (str): The public ed25519 key which was used to sign the object.
+            user_id (str): The user who owns the device.
+            device_id (str): The device who owns the key.
+
+        Returns:
+            True if the verification was successful, False if not.
+
+        Raises:
+            KeyError: Wrong json format.
```

Comment:
In case of wrong JSON format, let's just return `False`. This is already a checking function, and wrongly formatted JSON isn't any more or less valid than improperly signed JSON. This way, downstream apps don't have to guard calls to this method with try blocks.